pyo3-ffi/src/compat.rs

```diff
@@ -0,0 +1,44 @@
+//! C API Compatibility Shims
+//!
+//! Some CPython C API functions added in recent versions of Python are
+//! inherently safer to use than older C API constructs. This module
+//! exposes functions available on all Python versions that wrap the
+//! old C API on old Python versions and wrap the function directly
+//! on newer Python versions.
+
+// Unless otherwise noted, the compatibility shims are adapted from
+// the pythoncapi-compat project: https://github.com/python/pythoncapi-compat
+
+#[cfg(not(Py_3_13))]
+use crate::object::PyObject;
+#[cfg(not(Py_3_13))]
+use std::os::raw::c_int;
+
+#[cfg_attr(docsrs, doc(cfg(Py_3_13)))]
+#[cfg(Py_3_13)]
+pub use crate::dictobject::PyDict_GetItemRef;
+
+#[cfg_attr(docsrs, doc(cfg(not(Py_3_13))))]
```

Comment:
We need these cfgs to be empty else we end up with the wrong availability info here:

<img width="592" alt="image" src="https://github.com/user-attachments/assets/617fa671-182c-4b65-96d7-356fa166eef3">

It looks like using `all()` works:

```suggestion
#[cfg_attr(docsrs, doc(cfg(all())))]
#[cfg(Py_3_13)]
pub use crate::dictobject::PyDict_GetItemRef;

#[cfg_attr(docsrs, doc(cfg(all())))]
```

Reply:
Neat, here's how that looks
<img width="567" alt="image" src="https://github.com/user-attachments/assets/2dc56ca6-b0ff-4062-ac8b-cb3ffff9e7c8">

which is correct, if a little odd with the note about it always being available 🤣. I might see if rustdoc has an issue for that.

Reply:
Hmm interesting, when I tested locally I didn't see that. What nightly rust version did you try with?

Reply:
I got that doing `nox -s doc -- nightly internal` with `rustc 1.82-nightly (f8060d282 2024-07-30)` to generate the internal docs. It looks like a regular `nox -s doc` renders identically with no warning about the version under both CPython 3.13 and 3.12 though.

<img width="575" alt="image" src="https://github.com/user-attachments/assets/d4572623-d3eb-4b0c-89b7-ecae978bbaf6">

Reply:
Ah I think at least the `nightly` flag is needed for these hints, I thought I used it but maybe I got mixed up!